apps/annotations/views/tag_views.py

```diff
@@ -46,6 +47,19 @@ def get_success_url(self):
     def form_valid(self, form):
         form.instance.team = self.request.team
         form.instance.created_by = self.request.user
+        form.instance.name = unicodedata.normalize("NFC", form.instance.name)
+
+        existing_tags = Tag.objects.filter(
+            team=self.request.team,
+            name=form.instance.name,
+            is_system_tag=getattr(form.instance, "is_system_tag", False),
+            category=getattr(form.instance, "category", ""),
+        )
+
+        if existing_tags.exists():
+            form.add_error("name", "A tag with this name already exists for this team, system status, and category.")
+            return self.form_invalid(form)
+
         return super().form_valid(form)
```

Comment:
Could you not also wrap this in a try / catch and rely on the DB constraint for the duplicate check?

Reply:
[42ccbbd](https://github.com/dimagi/open-chat-studio/pull/1476/commits/42ccbbd8c4d3b88999c894ae2cd26ea0018fa910)